### PR TITLE
Disable canvas and compilation

### PR DESCRIPTION
**Pull request by frozenfrank**
This PR continues efforts begun in #328 by adding guards to newly added method calls. It also leverages the newly added `ApplicationProperties` system to read in the values from a properties file.

### CODE REVIEW

**Comment by frozenfrank** [comment]
Note that the commit message "RUN_COMPILATION can be controlled via ApplicationProperties" in commit 50cb2c6 is mostly a theoretical question. Nobody has shown me nor demonstrated where the `.properties` file exists so I'm not sure how to actually use it.

**Comment by 19mdavenport** [comment]
There is no .properties file associated with ApplicationProperties. It's tied to command line arguments

**Comment by frozenfrank** [comment]
> There is no .properties file associated with ApplicationProperties. It's tied to command line arguments

That explains it. I guess I just need some training on how to use the commandline arguments to adjust the values.

Is there something more you're looking for from this PR, or how can I help this one become approved as well?

**Review by 19mdavenport — CHANGES_REQUESTED** [review]
Could you add support for your new property to where it exists in main Server class, as well as providing it as an option in the docker compose file?

**Comment by frozenfrank** [comment]
> Could you add support for your new property to where it exists in main Server class, as well as providing it as an option in the docker compose file?

@19mdavenport I can look into the main `Server` class, but I have now idea how Docker works or while file I would be manipulating. I'll do my best.

**Comment by 19mdavenport** [comment]
https://github.com/softwareconstruction240/autograder/blob/main/compose.yml you'd just need to add it with the rest of the command line arguments

**Review by frozenfrank — COMMENTED** [review]
@19mdavenport I've added the support you asked for. 

(I still don't understand everything in that file, but I was able to follow the pattern to add support for the property. Thanks for the tip!)

File: src/main/java/edu/byu/cs/server/Server.java [hunk 1]
```diff
@@ -151,6 +151,9 @@ private static void setupProperties(String[] args) {
             if (cmd.hasOption("use-canvas")) {
                 properties.setProperty("use-canvas", cmd.getOptionValue("use-canvas"));
             }
+            if (cmd.hasOption("disable-compilation")) {
+                properties.setProperty("run-compilation", "false");
+            }
```
Comment (19mdavenport) [inline src/main/java/edu/byu/cs/server/Server.java:156]: Nice with the optional parameter

**Review by 19mdavenport — APPROVED** [review]

**Pull request merged**